core/src/datayoga_core/entity.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import importlib
+import logging
+import os
+import sys
+from abc import ABCMeta, abstractmethod
+from os import path
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+from datayoga_core.context import Context
+from datayoga_core.utils import get_bundled_dir, is_bundled, read_json
+from jsonschema import validate
+
+if TYPE_CHECKING:
+    from datayoga_core.block import Block
+    from datayoga_core.producer import Producer
+
+logger = logging.getLogger("dy")
+
+
+class Entity(metaclass=ABCMeta):
+    INTERNAL_FIELD_PREFIX = "__$$"
+    MSG_ID_FIELD = f"{INTERNAL_FIELD_PREFIX}msg_id"
+    RESULT_FIELD = f"{INTERNAL_FIELD_PREFIX}result"
+    OPCODE_FIELD = f"{INTERNAL_FIELD_PREFIX}opcode"
+    """
+    Block
+
+    Attributes:
+        properties Dict[str, Any]: Block properties
+    """
+
+    def __init__(self, properties: Optional[Dict[str, Any]] = None):
```

Comment:
they can't inherit from the same base class since Producer is about to become async. I suggest just putting the few shared lines of code into a module

Reply:
Reverted.